woocommerce/woocommerce
# Migrate Purchase task and dashboard utils to TS

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes #37190.

This PR migrates the Purchase task to TS. To archive that, I migrated the `dashboard/utils.js` as well.

Besides, this PR also fixes a wrong type definition (`productTypes`) in `@woocommerce/data` onboarding store.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

CI should pass

1. Go to the WooCommerce onboarding wizard.
2. Continue to `Onboarding wizard > Product Types` tab (/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=product-types)
3. Select a paid product type, such as `Memberships`
4. Complete OBW
5. Go to `WooCommerce > Home`
6. Observe that there is a `Add ... product to my store` task (for example: `Add Memberships and 1 more product to my store`) 
7. Click on that task.
8. Observe that a modal pop up with title `Would you like to add the following paid features to your store now?`
9. Click on `Buy now` button
10. It should redirect you to checkout page (https://woocommerce.com/checkout/...).
